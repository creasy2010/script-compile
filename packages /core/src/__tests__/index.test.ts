/**
 * @desc
 *
 * @使用场景
 *
 * @coder.yang2010@gmail.com
 * @Date    2020/7/15
 **/

import {compile} from "../index";

describe('基础编译', () => {
  it('基本转换', async () => {

    for (let i = 0, iLen = codes.length; i < iLen; i++) {
      let code1 = codes[i];
      let result  = compile(code1,{ast:true});
      debugger;
      expect(result.code).toMatchSnapshot("序号:"+i);
    }
  });
});

let codes =[
  "let a  =1;console.log(111)",
  'async function test (){let {data} = await api.get()}'
]
