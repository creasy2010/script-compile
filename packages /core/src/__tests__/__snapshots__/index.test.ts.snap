// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`基础编译 基本转换: 序号:0 1`] = `
"let a = 1;
console.log(111);"
`;

exports[`基础编译 基本转换: 序号:1 1`] = `
"async function test() {
  let {
    data
  } = await api.get();
}"
`;
